File: src/main/java/com/sedmelluq/discord/lavaplayer/container/matroska/format/MatroskaCuePoint.java
package com.sedmelluq.discord.lavaplayer.container.matroska.format;

/**
 * Matroska file cue point. Provides the offsets at a specific timecode for each track
 */
public class MatroskaCuePoint {
  /**
   * Timecode using the file timescale
   */
  public final long timecode;
  /**
   * Absolute offset to the cluster
   */
  public final long[] trackClusterOffsets;

  /**
   * @param timecode Timecode using the file timescale
   * @param trackClusterOffsets Absolute offset to the cluster
   */
  public MatroskaCuePoint(long timecode, long[] trackClusterOffsets) {
    this.timecode = timecode;
    this.trackClusterOffsets = trackClusterOffsets;
  }
}

File: main/src/main/java/lavaplayer/container/matroska/format/MatroskaCuePoint.kt
package lavaplayer.container.matroska.format

/**
 * Matroska file cue point. Provides the offsets at a specific timecode for each track
 *
 * @param timecode            Timecode using the file timescale
 * @param trackClusterOffsets Absolute offset to the cluster
 */
data class MatroskaCuePoint(
    /**
     * Timecode using the file timescale
     */
    @JvmField val timecode: Long,
    /**
     * Absolute offset to the cluster
     */
    @JvmField val trackClusterOffsets: LongArray
)
